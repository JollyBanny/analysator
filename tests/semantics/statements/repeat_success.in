var i : integer = 0;

begin
    repeat
        write(i);
        i += 1;
    until i > 5;
end.