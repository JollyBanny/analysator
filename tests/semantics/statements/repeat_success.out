program
└──── program block
      ├──── VAR
      │     └──── var declaration
      │           ├──── i
      │           ├──── integer
      │           └──── 0
      └──── Compound statement
            └──── REPEAT
                  ├──── >
                  │     ├──── i
                  │     └──── 5
                  ├──── Call statement
                  │     └──── write
                  │           └──── i
                  └──── +=
                        ├──── i
                        └──── 1

┌──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│  builtins                                                                                                            │
├──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│  Type             │  integer                                                                                         │
├──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│  Type             │  double                                                                                          │
├──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│  Type             │  char                                                                                            │
├──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│  Type             │  string                                                                                          │
├──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│  Type             │  boolean                                                                                         │
└──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
┌──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│  global                                                                                                              │
├──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│  Var              │  i                          │  integer                                                           │
└──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
